test/rules/immutable-name-snakecase.ts

```diff
@@ -0,0 +1,26 @@
+import assert from 'assert';
+import linter from '../utils/parsers';
+const contractWith = require('solhint/test/common/contract-builder').contractWith;
+
+const config = {
+    rules: { 'immutable-name-snakecase': 'error' },
+};
+
+describe('Linter - immutable-name-snakecase', () => {
+    it('should raise error for immutable name not in snakecase', () => {
```

Comment:
what about immutable private, will `_TEST_VAR` throw an error or not?

Reply:
it shouldn't; good idea. I've just added the test for that scenario